src/Microsoft.AspNet.Mvc.Core/AntiForgery/AntiForgeryWorker.cs

```diff
@@ -107,11 +111,11 @@ public TagBuilder GetFormInputElement([NotNull] HttpContext httpContext)
             SaveCookieTokenAndHeader(httpContext, newCookieToken);
 
             // <input type="hidden" name="__AntiForgeryToken" value="..." />
-            var retVal = new TagBuilder("input");
-            retVal.Attributes["type"] = "hidden";
-            retVal.Attributes["name"] = _config.FormFieldName;
-            retVal.Attributes["value"] = _serializer.Serialize(formToken);
-            return retVal;
+            var inputTag = new TagBuilder("input", _htmlEncoder);
+            inputTag.Attributes["type"] = "hidden";
```

Comment:
Super unrelated, but could we expose `Attributes` as an indexer on `TagBuilder`? We could do prettier initialization:

```C#
var inputTag = new TagBuilder("input", _htmlEncoder)
{
   ["type"] = "hidden",
   ...
}
```